### ISSUE TITLE
Seed cracking without the End dimension

### ISSUE THREAD

**Issue by hykilpikonna** [KaptainWutax/SeedCracker#12]
So my target server doesn't have an end dimension, is it possible to crack the seed only using the generated structures in the overworld dimension? It should be possible since it's just going through all seeds to find out which one has the same structure locations and stuff right?

**Comment by KaptainWutax** [comment]
It is possible but not on CPU. You'd need to get yourself a GPU program to go through the seeds, I believe Taleden has a program for it. I do not plan to ever offer support for GPU brute-forcing since it is hell to maintain.  

On another note, a friend of mine has a theory on how to get the entire seed just from a single dungeon. If it turns out to be reasonably fast, I will be implementing it in the Seed Cracker and you could then skip the end entirely.

So the answer to your question is effectively "yes but actually no." I'll keep you updated if any major breakthroughs happen. Thanks for your report!

**Comment by NintendoBest** [comment]
I'm also looking forward to this. I'm also have this issue that the server i want to find the seed does not have the end open to players. Hopefully your friend's theory will be ..."but actually yes" x)

**Comment by hykilpikonna** [comment]
> I'm also looking forward to this. I'm also have this issue that the server i want to find the seed does not have the end open to players. Hopefully your friend's theory will be ..."but actually yes" x)

There are actually many alternatives that I found that can achieve seed cracking with overworld only, but they require more effort. For example, if the seed is randomly generated, you can find about 18 slime chunks using in-game chunk digging methods and then use the slime chunk seed cracker (https://github.com/pruby/slime-seed) to find the 48-bit seed and then convert it to the 64-bit world seed on the bottom of this site: (https://badel2.github.io/slime_seed_finder/).

And you can also try Neil's GPU brute-forcing with structure locations (https://github.com/hube12/GPUExample), but it's not documented so I didn't test it.

Also, I'm working on a seed cracker that works with all available information so that no matter what info is given, it can at least output the remaining candidates, but try the methods above first because my project would just be a combination of the methods described above.

**Comment by KaptainWutax** [comment]
I am aware of those methods and have written several GPU crackers in the past. This mod is strictly CPU though, and I plan to keep it so. There is currently no seed cracker out there that can reverse-engineer the 48 bits in a reasonable amount of time on CPU but many concepts are being worked on.

**Comment by KaptainWutax** [comment]
A lattice tool has been developed by Matthew that should make dungeon cracking super fast on CPU: https://github.com/mjtb49/JavaRandomReverser. Huge milestone for the implementation of the feature.

**Comment by KaptainWutax** [comment]
With version v0.1.0-beta and onwards, this is now supported through the dungeon finder. Make sure to enable it first since I have disabled it until I'm sure it's stable.

**Comment by Zorro909** [comment]
It should be possible to also crack the complete world Seed without the End dimension, right?
At least I've got myself the Structure Seed (the lower 48 bits) via your mod. [With two tries, I could eliminate every seed, but one] and then I used this library https://github.com/Cubitect/cubiomes to guess the seed from the remaining ~65000 (2^16) seeds using 10 biomes. Since this mod already gathers the biome data, this should be possible to implement, right? (I'd also be open to help out)
